lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2221,7 +2221,21 @@ namespace {
               MD->overwriteAccess(AccessLevel::Private);
             } else if (cxxOperatorKind ==
                        clang::OverloadedOperatorKind::OO_PlusPlus) {
-              if (cxxMethod->param_empty()) {
+              // Make sure the type is not an immortal foreign reference type.
+              // We cannot handle `operator++` for those types, since the
+              // current implementation creates a new instance of the type.
```

Comment:
Creating a new instance of any foreign reference type will have this issue, not just immortal foreign reference types. 

This change can just be:
```
if (cxxMethod->param_empty() && !isa<ClassDecl>) {
```